File: 1079-letter-tile-possibilities/1079-letter-tile-possibilities.cpp
class Solution {
public:
    int numTilePossibilities(string tiles) {
        int ans = 0;
        vector<int> count(26, 0);
        for (char c : tiles) {
            count[c - 'A']++;
        }
        function<void()> solve = [&]() {
            ans++;
            for (int i = 0; i < 26; i++) {
                if (count[i] == 0) continue;
                count[i]--;
                solve();
                count[i]++;
            }
        };
        solve();
        return ans - 1;
    }
};

File: LeetCode/1079-letter-tile-possibilities/1079-letter-tile-possibilities.py
class Solution:
    def numTilePossibilities(self, tiles: str) -> int:
        from functools import lru_cache

        count = [0] * 26
        for c in tiles:
            count[ord(c) - ord('A')] += 1

        self.ans = 0

        def solve():
            self.ans += 1
            for i in range(26):
                if count[i] == 0:
                    continue
                count[i] -= 1
                solve()
                count[i] += 1

        solve()
        return self.ans - 1